src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,157 @@
+  } else {
+    const expiredList = itfList.filter(itf => itf.status === PRESTART_STATUSES.expired);
+    const activeList = itfList.filter(itf => itf.status === PRESTART_STATUSES.active);
+    // If the user doesn't have any active or expired ITFs, set none status        
+    if (expiredList.length === 0 && activeList.length === 0) {
+      status = PRESTART_STATUSES.none;
+      // If the user doesn't have any active ITFs, set expired status
+    } else if (activeList.length === 0) {
+      status = PRESTART_STATUSES.expired;
+      expirationDate = getLatestTimestamp(expiredList.map(itf => itf.expirationDate));
+      dispatch(setPrestartData({ previousExpirationDate: expirationDate }));
+      // If the user has an active ITF, set retrieved status
+    } else {
+      status = PRESTART_STATUSES.retrieved;
+      expirationDate = activeList[0].expirationDate;
+      dispatch(setPrestartData({ currentExpirationDate: expirationDate }));
+    }
+  }
+  dispatch(setPrestartStatus(status));
+  return status;
+};
+
+export const handleCheckFailure = (error, hasSavedForm, dispatch) => {
+  const status = hasSavedForm ? PRESTART_STATUSES.notRetrievedSaved : PRESTART_STATUSES.notRetrievedNew;
+  dispatch(setPrestartStatus(status));
+  return status;
+};
+
+export function checkITFRequest(dispatch, hasSavedForm) {
+  return apiRequest(
+    '/intent_to_file',
+    null,
+    ({ data }) => handleCheckSuccess(data, dispatch),
+    ({ errors }) => handleCheckFailure(errors, hasSavedForm, dispatch)
+  );
+}
+
+const delay = (func, wait, args) => setTimeout(() => func(args), wait);
+const fakeITFRequest = (url, options, success) => delay(success, 1000, existingITFData);
+
+export function mockCheckITFRequest(dispatch, hasSavedForm) {
+  return fakeITFRequest(
+    '/intent_to_file',
+    null,
+    ({ data }) => handleCheckSuccess(data, dispatch),
+    ({ errors }) => handleCheckFailure(errors, hasSavedForm, dispatch)
+  );
+}
+
+export const handleSubmitSuccess = (data, successStatus, dispatch) => {
+  const expirationDate = data.attributes.intentToFile.expirationDate;
+  dispatch(setPrestartData({ currentExpirationDate: expirationDate }));
+  dispatch(setPrestartStatus(successStatus));
+};
+
+export const handleSubmitFailure = (error, errorStatus, dispatch) => {
+  dispatch(setPrestartStatus(errorStatus));
+};
 
+export function submitITFRequest(dispatch, successStatus, errorStatus) {
+
+  return apiRequest(
+    '/intent_to_file/compensation',
+    { method: 'POST' },
+    ({ data }) => handleSubmitSuccess(data, successStatus, dispatch),
+    ({ errors }) => handleSubmitFailure(errors, errorStatus, dispatch)
+  );
+}
+
+export function verifyIntentToFile(hasSavedForm) {
+  return async (dispatch) => {
+    let submitSuccessStatus;
+    let submitErrorStatus;
+    dispatch(setPrestartStatus(PRESTART_STATUSES.pending));
+
+    const existingITFStatus = await checkITFRequest(dispatch, hasSavedForm);
```

Comment:
`checkITFRequest` first dispatches `setPrestartStatus(status)`, then returns `status`. The returned `status` string is then used to derive a payload for `submitITFRequest` within the `verifyIntentToFile` action creator.

My concern with this setup is that it might be possible to have inconsistencies in state crop up, due to the fact that we're both dispatching actions and at the same time using values returned from those action creators in other action creators that themselves make async calls using the same data. I'd be curious to hear @cvalarida 's thoughts on this as well.

Reply:
Sounds good, `setPrestartStatus` moved into `verifyIntentToFile`.